qiskit_experiments/library/characterization/T2Hahn.py

```diff
@@ -0,0 +1,116 @@
+from typing import Union, Iterable, Optional, List
+
+import numpy as np
+from numpy.random import Generator, default_rng
+
+from qiskit import QuantumCircuit, QiskitError, QuantumRegister, ClassicalRegister
+from qiskit.ignis.characterization.characterization_utils import pad_id_gates, time_to_ngates
+from qiskit.providers import Backend
+from qiskit.quantum_info import Clifford
+from qiskit.providers.options import Options
+from qiskit.circuit import Gate
+
+from qiskit_experiments.framework import BaseExperiment, ParallelExperiment
+from qiskit_experiments.curve_analysis.data_processing import probability
+from .rb_analysis import RBAnalysis
+from .clifford_utils import CliffordUtils
+from .rb_utils import RBUtils
+
+
+class T2Hahn(BaseExperiment):
+
+    # Analysis class for experiment
+    __analysis_class__ = T2Analysis  # need to add T2Analysis
+
+    def __init__(
+            self,
+            qubits: Union[int, Iterable[int]],
+            lengths: Union[List[int], np.array],  # need to change name?
+            gate_time: float,
+            n_echos: int = 1,
```

Comment:
Hahn echo is always `n_echoes=1`. You need to create dynamical decoupling base class, then create HahnEcho on top of it. Note that there are various patterns to implement echoes, i.e. different combination of pulses (X-X or X-Y), different interval patterns (UDD, PDD, etc...), and combination of them. So the base class should be enough flexible to implement all of them. Then create experiment class for each specific realization. T2Ramsey can also be one of realization of dynamical decoupling with n echo = 0.

https://scirate.com/arxiv/1012.4262

Reply:
We should do it gradually. For now only a Hahn echo experiment, with a single echo and one hard-coded combination of pulses, inheriting directly from `BaseExperiment`. Later, if we want (depending on importance and priorities), we can add more experiments (including construction of hierarchy between experiments) and options within an experiment.

Reply:
Doing it later would be fine. but the API of Hahn echo should be the same with the T2Ramsey.

Reply:
Yes, except for the frequency parameter, that I don't want to introduce (yet) for the Hahn echo experiment.

Reply:
Yes, that's true.